src/internal.cc

```diff
@@ -256,13 +256,28 @@ namespace rego
     return in_query(node->parent()->shared_from_this());
   }
 
-  Node err(NodeRange& r, const std::string& msg, const std::string& code)
+  // FIXME, if we make this concurrent, then this should be atomic.
+  static bool error_status{false};
```

Comment:
Is there a way we can make this per-interpreter? Perhaps pass an error object when running the pass (as is done with built-ins presently?)